asmcomp/printcmm.ml

```diff
@@ -150,7 +150,7 @@ let rec expr ppf = function
     fprintf ppf "block-hdr(%s)%s"
       (Nativeint.to_string n) (location d)
   | Cconst_float (n, _dbg) -> fprintf ppf "%F" n
-  | Cconst_symbol (s, _dbg) -> fprintf ppf "\"%s\"" s
+  | Cconst_symbol (s, ty, _dbg) -> fprintf ppf "\"%s\" %a" s machtype ty
```

Comment:
You should probably put parentheses around the whole expression, or stick the type to the string in some way, otherwise the printing of a list of arguments including symbols gets confusing.